### PR TITLE
Support Django 1.9

### PR DESCRIPTION
**Pull request by giginet**

### CODE REVIEW

**Comment by lambdalisue** [comment]
Thanks. I'll follow rest of this.

**Comment by giginet** [comment]
In Django 1.9, `add_to_builtin` function have been removed. 
https://docs.djangoproject.com/en/1.9/releases/1.9/#django-template-base-add-to-builtins-is-removed

So I guess to need radical changes.

File: src/permission/conf.py [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
 from django.core.exceptions import ImproperlyConfigured
 from appconf import AppConf
 from permission.handlers import LogicalPermissionHandler
 
 
+__all__ = ('settings',)
+
+
 class PermissionConf(AppConf):
+
```
Comment (giginet) [inline src/permission/conf.py]: PEP8

File: src/permission/tests/test_templatetags/test_permissionif.py [hunk 2]
```diff
@@ -243,11 +242,203 @@ def setUp(self):
...
+            "'permission.permission_templatetag_test_perm1' "
+            "and user has 'permission.permission_templatetag_test_perm2' %}"
+            "Success"
+            "{% endpermission %}"
+        ).render(context)
+
+        self.assertEqual(out, "Success")
+
+    def test_permissionif_tag_or(self):
+
```
Comment (giginet) [inline src/permission/tests/test_templatetags/test_permissionif.py]: unnecessary

Comment (lambdalisue) [inline src/permission/tests/test_templatetags/test_permissionif.py]: It stands for testing *buitlins* rather than *override*. So it is required ;-)

Comment (lambdalisue) [inline src/permission/tests/test_templatetags/test_permissionif.py]: You were talking about PEP8. OK lol.

File: src/permission/__init__.py [hunk 3]
```diff
@@ -12,8 +8,43 @@
...
-if settings.PERMISSION_REPLACE_BUILTIN_IF:
-    from permission.compat import add_to_builtins
-    add_to_builtins('permission.templatetags.permissionif')
+from django.core.exceptions import ImproperlyConfigured
+
+if 'permission' in settings.INSTALLED_APPS:
+    if settings.PERMISSION_CHECK_AUTHENTICATION_BACKENDS:
+        if ('permission.backends.PermissionBackend' not in
+                settings.AUTHENTICATION_BACKENDS):
+            raise ImproperlyConfigured(
```
Comment (giginet) [inline src/permission/__init__.py]: :eyes:

File: tox.ini [hunk 4]
```diff
@@ -19,28 +20,22 @@ deps=
     django16: django>=1.6,<1.7
     django17: django>=1.7,<1.8
     django18: django>=1.8,<1.9
+    django19: django>=1.9,<2.0
```
Comment (giginet) [inline tox.ini]: Next version will be `1.10.0`

https://www.djangoproject.com/weblog/2015/jun/25/roadmap/

File: tox.ini [hunk 5]
```diff
@@ -19,28 +20,22 @@ deps=
...
-    py33,py34: cp
-    py33,py34: 2to3
+    {envbindir}/coverage run --append --source=src/permission runtests.py []
 
 [testenv:docs]
 basepython=python
 changedir=docs
-deps=-rrequirements-docs.txt
+deps=
+    django<1.9
```
Comment (giginet) [inline tox.ini]: Is is necessary?

File: src/permission/tests/test_templatetags/test_permissionif.py [hunk 6]
```diff
@@ -243,11 +242,203 @@ def setUp(self):
         self._original_registry = registry._registry
         # clear registry and register mock handler
         registry._registry = {}
+        # Make sure builtin if is not replaced
```
Comment (giginet) [inline src/permission/tests/test_templatetags/test_permissionif.py]: Is it correct?

**Comment by giginet** [comment]
LGTM with nits

File: tox.ini (hunk 4 above)
Comment (lambdalisue) [inline tox.ini]: True.

File: tox.ini (hunk 5 above)
Comment (lambdalisue) [inline tox.ini]: True.

File: src/permission/tests/test_templatetags/test_permissionif.py (hunk 6 above)
Comment (lambdalisue) [inline src/permission/tests/test_templatetags/test_permissionif.py]: True.

File: src/permission/__init__.py (hunk 3 above)
Comment (lambdalisue) [inline src/permission/__init__.py]: What did you mean?

Comment (giginet) [inline src/permission/__init__.py]: I checked

**Comment by lambdalisue** [comment]
Thanks. I'll merge it ;-)

File: src/permission/tests/test_utils/__init__.py [hunk 7]
```diff
@@ -1,4 +1,8 @@
...
-from test_permissions import *
-from test_field_lookup import *
+import django
+if django.VERSION < (1, 6):
+    # Django 1.5 and earlier use a different test method and the followings
+    # are required to be exposed
+    from permission.tests.test_utils.test_logics import *
+    from permission.tests.test_utils.test_handlers import *
+    from permission.tests.test_utils.test_permissions import *
+    from permission.tests.test_utils.test_field_lookup import *
```
Comment (rakhi27) [inline src/permission/tests/test_utils/__init__.py]: How can we import if we use django 1.9?

Comment (lambdalisue) [inline src/permission/tests/test_utils/__init__.py]: In Django 1.9, test runner automatically detect test cases so you don't need to import explicitly.

**Pull request merged**